### ISSUE TITLE
Provide Feedback when cabal-helper is being built

### ISSUE THREAD

**Issue by alanz** [haskell/haskell-ide-engine#913]
For a first time startup with a given GHC version, cabal-helper setup does a lot of building, as it installs a version of Cabal among other things.

This means that with a fresh HIE install, the user experience is to start it up, open a project, and have the CPU silently max out for a significant number of minutes.

We should provide a `window/showMessage` to explain what is going on, and perhaps provide progress reports along the way, the way the rust language server does.

**Comment by thomashoneyman** [comment]
I believe I've run into the same thing. I installed the language server manually and then opened up a project in VSCode expecting to see the result, as is typical with other tools of this sort that I use (PureScript IDE, etc.), but thought that the installation was bad before noticing all of this stuff being built:

<img width="1262" alt="screen shot 2019-01-17 at 11 37 05 pm" src="https://user-images.githubusercontent.com/10245104/51372135-0b24b780-1ab1-11e9-9c40-2ec3bca73cbf.png">

I don't mind waiting for that to complete, but I spent a bit of time thinking the plugin just wasn't working. It would be much better to receive some sort of notification that this is in progress, as you noted.
